src/biokbase/narrative/jobs/job.py

```diff
@@ -77,65 +117,104 @@ def __init__(self, **kwargs):
-        else:
-            self.job_id = kwargs.get("job_id")
-
-        # parent job: set the batch_id
-        if kwargs.get("batch_job", False) is True:
-            kwargs["batch_id"] = self.job_id
-
-        for key, value in JOB_DEFAULTS.items():
-            setattr(self, key, kwargs.get(key, value))
+        # reconstruct the ee2 job state object
+        ee2_state = kwargs.get("ee2_state", {})
+        job_input = ee2_state.get("job_input", {})
+        narr_cell_info = job_input.get("narrative_cell_info", {})
 
-        ee2_state = kwargs.get("ee2_state", None)
-        if not ee2_state:
-            ee2_state = {
-                "job_id": self.job_id,
+        narr_cell_info.update(
+            {key: kwargs[key] for key in NARR_CELL_INFO_ATTRS if key in kwargs and key not in narr_cell_info}
+        )
+        job_input.update(
+            {
+                **{_attr_to_ee2(key): kwargs[key] for key in JOB_INPUT_ATTRS if key in kwargs and key not in job_input},
+                "narrative_cell_info": narr_cell_info,
+            }
+        )
+        ee2_state.update(
+            {
+                **{key: kwargs[key] for key in STATE_ATTRS if key in kwargs and key not in ee2_state},
+                "job_input": job_input
             }
-            # reconstruct the ee2 job state object
-            for key in ["batch_id", "batch_job", "child_jobs", "owner"]:
-                if key in kwargs:
-                    ee2_state[key] = kwargs[key]
+        )
 
-        self.update_state(ee2_state)
+        return cls(
+            ee2_state=ee2_state,
+            extra_data=kwargs.get("extra_data"),
+            children=kwargs.get("children")
+        )
 
-    @classmethod
-    def from_state(cls, job_state):
+    def __getattr__(self, name):
         """
-        Parameters:
-        -----------
-        job_state - dict
-            the job information returned from ee2.check_job
-
+        Map expected job attributes to paths in stored ee2 state
         """
-        job_input = job_state.get("job_input", {})
-        narr_cell_info = job_input.get("narrative_cell_info", {})
-        return cls(
-            app_id=job_input.get("app_id", JOB_DEFAULTS["app_id"]),
-            app_version=job_input.get("service_ver", JOB_DEFAULTS["app_version"]),
-            batch_id=job_state.get("batch_id", JOB_DEFAULTS["batch_id"]),
-            batch_job=job_state.get("batch_job", JOB_DEFAULTS["batch_job"]),
-            cell_id=narr_cell_info.get("cell_id", JOB_DEFAULTS["cell_id"]),
-            child_jobs=job_state.get("child_jobs", JOB_DEFAULTS["child_jobs"]),
-            ee2_state=job_state,
-            job_id=job_state.get("job_id"),
-            owner=job_state.get("user", JOB_DEFAULTS["owner"]),
-            params=job_input.get("params", JOB_DEFAULTS["params"]),
-            run_id=narr_cell_info.get("run_id", JOB_DEFAULTS["run_id"]),
-            tag=narr_cell_info.get("tag", JOB_DEFAULTS["tag"]),
+        attr = dict(
```

Comment:
Long lines

Reply:
`black` or `flake8` (probably black) may help with that if you haven't run it already.

Reply:
Wow black is cool.
Except they wanted to do this:
```
            run_id=lambda: self._const_state.get("job_input", {})
            .get("narrative_cell_info", {})
            .get("run_id", JOB_ATTR_DEFAULTS["run_id"]),
            tag=lambda: self._const_state.get("job_input", {})
            .get("narrative_cell_info", {})
            .get("tag", JOB_ATTR_DEFAULTS["tag"]),
            user=lambda: self._const_state.get("user", JOB_ATTR_DEFAULTS["user"]),
        )
```
But otherwise nice.

Reply:
Ugh, really? That's not what I expected. An indent or two there would be nice. Ah well.